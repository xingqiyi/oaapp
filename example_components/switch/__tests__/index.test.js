import React from 'react';
import { render, shallow } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Switch from '../index';

describe('SegmentedControl', () => {
  it('renders correctly', () => {
    const wrapper = render(<Switch checked />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('check api', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Switch checked onChange={onChange} />);
    wrapper.find('input').simulate('change', { target: { checked: false } });
    expect(onChange).toHaveBeenCalledWith(false);
  });
});
