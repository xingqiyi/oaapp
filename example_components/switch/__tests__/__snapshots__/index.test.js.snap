// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl renders correctly 1`] = `
<label
  class="am-switch"
>
  <input
    checked=""
    class="am-switch-checkbox"
    name=""
    type="checkbox"
    value="on"
  />
  <div
    class="checkbox"
  />
</label>
`;
